book/website/reproducible-research/code-quality/code-quality-robust.md

```diff
@@ -0,0 +1,170 @@
+(rr-code-error)=
+# Writing robust code
+
+We all have experienced it: you just wrote a new piece of code, but when you try it, it does not work as expected.
+Perhaps there is a typo, a bug, or you just passed the wrong parameter to a function.
+It is not a big deal... as long as you notice.
+Noticing something is wrong is the first step to fixing it.
+
+## Silent failures
+There will be many assumptions you make when writing a program.
+For example, the data type of your imports, the structure of a data file, but also the behavior of any dependencies, from individual functions, entire libraries, to the programming language you use and how it functions in various operating systems.
+It is natural to have assumptions to build on, but it can become problematic when these assumptions are incorrect for a specific instance, yet the program carries on regardless.
+This is what is called a 'silent failure'.
+
+Silent failures can lead to problems down the line, likely resulting in strange and unintelligible error messages that do not have anything to do with the actual problem.
+A silent failure that stays silent throughout will generate results that are wrong, and needs sharp eyes to be detected.
+To make sure these things do not happen, your program needs built-in checks and balances.
+Having good error management practices dramatically reduces the chance that a problem occurs, and especially, that it passes unnoticed.
+This chapter's main purpose is to help you make your code robust, and capable of dealing with different potential problems.
+
+(rr-code-error-workflow)=
+## Workflow
+
+(rr-code-error-workflow-step1)=
+### Step 1: Describe your assumptions
+
+Your code contains many assumptions.
+For example, a function performing a simple mathematical operation assumes that its input is numerical.
+What happens if this function is used on a bit of text?
+Or a dataframe?
+Or an open file?
+
+In a different example, let's imagine a data science workflow.
+As part of this workflow, a column labeled "Age" is selected.
+What happens if this column does not exist?
+The workflow plots the column "Age", but the plot is cut off at age 100.
+What happens if the data contains ages over 100?
+Or if this column contains negative numbers?
+
+In making decisions about cases like these, the first step is to be explicit about the assumptions made.
+To identify assumptions in your code, you can ask yourself:
+
+- What type of object/data do I expect here?
+- What files need to exist for my workflow to run, and where?
+- When calling a function, am I relying on default settings for arguments?
+- When a function returns multiple outputs, do I ensure they are in the right order?
+
+If you think critically, there is no end to the assumptions that you make.
+For instance, you assume that a built-in function you use works in a specific way.
+Labeling all these assumptions would perhaps drive you crazy, and that is not the point of this exercise.
```

Comment:
```suggestion
Labeling all these assumptions would perhaps keep you busy eternally, and that is not the point of this exercise.
```

Or another phrasing - crazy is ableist language that we do not use in The Turing Way.